### PR TITLE
[MKLDNN] Fuse Conv+BatchNorm + ReLU

### PR DESCRIPTION
**Pull request by Sand3r-**
Fuses Convolution with ReLU using transpiler when MKLDNN is used.

Also, due to small re-arrangement in transpiler methods execution order, now the whole [Conv] -> [BN] -> [ReLU] is fused into a single [Conv].

On ResNet-50, this change speeds up the whole topology on `OMP_NUM_THREADS=1` and `batch_size=1` by **17%**. and **33%** on CRNN-CTC.

### CODE REVIEW

File: python/paddle/fluid/transpiler/inference_transpiler.py [hunk 1]
```diff
@@ -60,13 +60,46 @@ def transpile(self, program, place, scope=None):
...
+        :type program: Program
+        '''
+        self.block = program.block(0)
+
+        i = 0
+        while i < len(self.block.ops):
+            current_op = self.block.ops[i]
+            if current_op.type in ['conv2d']:
+                next_op = self.block.ops[i + 1]
+                if next_op.type == 'relu':
```
Comment (jczaja) [inline python/paddle/fluid/transpiler/inference_transpiler.py:90]: Wouldn't it be more reliable to check if further (not necessarily next ) operator is using convolution output as its input and this op is relu ?

Comment (Sand3r-) [reply python/paddle/fluid/transpiler/inference_transpiler.py:90]: Since the fusing will soon be moved to Pass mechanism, the transpiler version was merely meant to serve as a tool allowing to test whether the fuse works and how well does it perform.

The requested change is certainly reasonable when considering the topologies where groups occur, however finding the layer which takes convolution output as input might be a bit problematic given that inputs of various operators have different names (sometimes it is "Input", other times "In", or "X" in case of two-argument operators) under which the actual tensor names are stored. Moreover all of the existing input tensors of all operators would have to be tested against an output tensor name.

In other words, performing this check would end up being just like the fused operator name indicates - convoluted.

Comment (jczaja) [reply python/paddle/fluid/transpiler/inference_transpiler.py:90]: ok

**Review by jczaja — APPROVED** [review]
LGTM
